src/dss/utils.py

```diff
@@ -211,3 +228,89 @@ def get_labels_for_node(lightkube_client: Client) -> dict:
+    """
+    Truncate the image component to ensure the total length of name, image,
+    and url including delimiters does not exceed the maximum length.
+
+    Args:
+        name (str): The name component of the row.
+        image (str): The image component of the row.
+        url (str): The URL component of the row.
+        max_length (int, optional): The maximum total length allowed for the
+                                    row including delimiters. Defaults to 80.
+
+    Returns:
+        tuple[str, str, str]: The components of the row (name, image, url)
+                              after adjusting the length.
+    """
+    # Account for table characters: 4 vertical bars and 6 spaces (2 per column for padding)
+    delimiter_length = 4 + 6
+    content_max_length = max_length - delimiter_length
+
+    current_length = len(name) + len(image) + len(url)
+    if current_length > content_max_length:
+        excess_length = current_length - content_max_length
+        new_image_length = len(image) - excess_length
+        if new_image_length > 3:
+            image = image[: new_image_length - 3] + "..."
+        else:
+            image = "..."
+    return name, image, url
+
+
+def get_deployment_state(deployment: Deployment, lightkube_client: Client) -> DeploymentState:
+    """
+    Determine the state of a Kubernetes deployment, which is constrained to 0 or 1 replicas.
+
+    Args:
+        deployment (Deployment): The deployment object.
+        lightkube_client (Client): The Kubernetes client.
+
+    Returns:
+        DeploymentState: The state of the deployment as an enumeration.
+    """
+    # Extract replica counts and deletion timestamp from deployment
+    desired_replicas = deployment.spec.replicas if deployment.spec.replicas is not None else 0
+    current_replicas = deployment.status.replicas if deployment.status.replicas is not None else 0
+    deletion_timestamp = deployment.metadata.deletionTimestamp
+
+    # Check for deployment deletion
+    if deletion_timestamp:
+        return DeploymentState.REMOVING
+
+    # Check pod statuses for image pulling issues and container creation
+    pods = lightkube_client.list(
+        Pod, namespace=deployment.metadata.namespace, labels=deployment.spec.selector.matchLabels
+    )
+    for pod in pods:
+        for container_status in pod.status.containerStatuses:
+            if container_status.state.waiting:
+                if container_status.state.waiting.reason in [
+                    "ImagePullBackOff",
+                    "ErrImagePull",
+                    "ContainerCreating",
+                ]:
+                    return DeploymentState.DOWNLOADING
+
+    # Determine the state based on replica counts
+    if desired_replicas == 0:
+        if current_replicas == 0:
+            return DeploymentState.STOPPED
+        else:
+            return DeploymentState.STOPPING
+    else:  # desired_replicas == 1
```

Comment:
I think this works as is, but I'm not 100% because the assumption is being made that `desired_replicas` is always either 0 or 1.  If by some accident `desired_replicas = 2`, I'm not 100% sure what will happen here.

Rather than doing `else: # desired_replicas == 1` and assuming it can never be 2, can we handle that explicitly?  Something like:
```python
if desired_replicas == 0:
  ...
elif desired_replicas == 1:
  ...
else:
  # error!
```

that way there's no ambiguity